src/Compilers/CSharp/Test/Emit2/FlowAnalysis/FlowTests.cs

```diff
@@ -5635,12 +5647,138 @@ class Program
                     static void Main()
                     {
                         const int N = 10;
+                        const int Unused = 20;
                         var lam = ([A(Prop = N)] int x) => { };
                         lam(100);
                     }
                 }
                 """;
-            CreateCompilation(source).VerifyDiagnostics();
+            CreateCompilation(source).VerifyDiagnostics(
+                // (12,19): warning CS0219: The variable 'Unused' is assigned but its value is never used
+                //         const int Unused = 20;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "Unused").WithArguments("Unused").WithLocation(12, 19));
+        }
+
+        [Fact, WorkItem(60645, "https://github.com/dotnet/roslyn/issues/60645")]
+        public void LocalMethod_AttributeArguments()
+        {
+            var source = """
+                using System;
+                class A : Attribute
+                {
+                    public A(int param) { }
+                    public int Prop { get; set; }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        const int N1 = 10;
+                        const int N2 = 20;
+                        const int N3 = 30;
+                        const int N4 = 40;
+                        const int N5 = 50;
+                        const int N6 = 60;
+                        [A(N1, Prop = N2)][return: A(N3, Prop = N4)] int F() => N5;
```

Comment:
Although this PR is focusing on attributes on the lambda/local function, consider adding similar tests for parameters of lambdas/local functions.

Reply:
Such tests were already added in the previous PR:

https://github.com/dotnet/roslyn/pull/64660/files#diff-fc05250f65f4e4fed831dc87cf19115cf1a315d3bfc07d47fa1300a4864fc769